### PR TITLE
Odc

### PR DESCRIPTION
**Pull request by CMonnin**
refactored a lot of code

- now can work on ODC playground
  - images need to be hosted somewhere
- can take json or csv as configuration files
- zod validation implemented

### CODE REVIEW

File: src/globalState.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export let useJson = false;
+
+export function setUseJsonToTrue(): void {
```
Comment (joshunrau) [inline src/globalState.ts]: getter and setter (or normal method, depending on preference)

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import type { Language } from "/runtime/v1/@opendatacapture/runtime-core";
+
+import "/runtime/v1/jspsych@8.x/css/jspsych.css";
+
+import { defineInstrument } from "/runtime/v1/@opendatacapture/runtime-core";
+
+if (!useJson) {
+  setUseJsonToTrue();
+}
```
Comment (joshunrau) [inline src/index.ts:16]: mistake

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  content: {
+    render(done) {
+      pictureNamingTask(done);
+    },
+  },
+  details: {
+    description: "A jsPysch implementation of the Boston Naming Task",
+    estimatedDuration: 1,
+    instructions: ["<PLACEHOLDER>"],
+    license: "UNLICENSED",
```
Comment (joshunrau) [inline src/index.ts]: make sure to set to Apache 2.0 if possible

File: src/pictureNamingTask.ts [hunk 4]
```diff
@@ -1,126 +1,178 @@
...
-
-/*
+import { transformAndDownload, transformAndExportJson } from "./dataMunger.ts";
+import { experimentSettingsJson } from "./experimentSettings.ts";
+import { experimentSettingsCSV, imageDbCSV } from "./fetchAndParse.ts";
+import { useJson } from "./globalState.ts";
+import i18n from "./i18n.ts";
+import {
+  $experimentImage,
+  $settings,
```
Comment (joshunrau) [inline src/pictureNamingTask.ts]: Follow naming convention $Schema

File: src/schemas.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import { type SupportedLanguage } from "./i18n.ts";
+
+import { z } from "/runtime/v1/zod@3.23.x";
```
Comment (joshunrau) [inline src/schemas.ts:1]: naming convention

**Review by joshunrau — CHANGES_REQUESTED** [review]

File: src/schemas.ts (hunk 5 above)
Comment (CMonnin) [reply src/schemas.ts:1]: see 9d71f92

File: src/pictureNamingTask.ts (hunk 4 above)
Comment (CMonnin) [reply src/pictureNamingTask.ts]: see 9d71f92

File: src/index.ts (hunk 2 above)
Comment (CMonnin) [reply src/index.ts:16]: see 3eb3e94

File: src/globalState.ts (hunk 1 above)
Comment (CMonnin) [reply src/globalState.ts]: see 3eb3e94

File: src/index.ts (hunk 3 above)
Comment (CMonnin) [reply src/index.ts]: see 90f0af9

**Pull request merged**